data_collection/web/guides.rst

```diff
@@ -221,187 +221,184 @@ indicate no value.
+          category: ["Crafts & Sewing", "Toys"],
+          price: "49.90",
+          quantity: 3,
+          brand: "DMZ",
+          variant: "blue"
+      }
+  ]);
 
 .. _guide_tracking_cart_updates:
 
 Tracking cart updates
 ^^^^^^^^^^^^^^^^^^^^^
 
-Another type of e-commerce activity you can track is an update of a shopping cart. With it, we are able to measure how often visitors don't complete the ordering process and what products stay in abandoned carts.
-
-Tracking a cart update has two steps: registering items from the cart and sending them. The following example uses two functions - :ref:`addEcommerceItem<jtc-api-addEcommerceItem>` and :ref:`trackEcommerceCartUpdate<jtc-api-trackEcommerceCartUpdate>` - to achieve exactly that.
-
-.. code-block:: javascript
-
-    // visitor added one chocolate bar to an empty shopping cart
-
-    // register chocolate bar with...
-    _paq.push(["addEcommerceItem",
-        "82775027",                 // SKU (stock-keeping unit)
-        "MEGA Milk Chocolate 200g", // name
-        "Candy",                    // category
-        6.00,                       // price
-        1                           // quantity
-    ]);
-
-    // track cart update with a total value of 6.00
-    _paq.push(["trackEcommerceCartUpdate", 6.00]);
-
-This code snippet sends a cart update event with a cart containing one item (SKU *candy-12837*, name *MEGA Milk Chocolate 200g*, category *Candy*, price *6.00*) and having total value of *6.00*.
-
-The list of registered items is stored only in memory. **Reloading the page will clear the list** and the previously registered items will have to be added again.
-
-.. code-block:: javascript
-
-    // visitor added one mango fruit to a shopping cart with one chocolate bar
-
-    // register previously added items
-    _paq.push(["addEcommerceItem", "82775027", "MEGA Milk Chocolate 200g", "Candy", 6.00, 1]);
-
-    // register the new item
-    _paq.push(["addEcommerceItem", "01809926", "FRUTASTIC Mango", "Fruits & vegetables", 4.00, 1]);
-
-    // track cart update with a total value of 10.00
-    _paq.push(["trackEcommerceCartUpdate", 10.00]);
-
-.. note::
-
-    If you are not sure what items have been registered, use :ref:`getEcommerceCart<jtc-api-getEcommerceItems>` function.
-
-    .. code-block:: javascript
-
-        _paq.push([function() { console.log(this.getEcommerceItems()); }]);
+Another type of e-commerce activity you can track is an update of a shopping cart.
+With it, we are able to measure how often visitors add or remove specific products to the cart and what products stay in abandoned carts.
 
-Because single page applications do not refresh the page when a visitor manipulates the cart, an e-commerce implementation in SPAs must either:
-
-1. Clear the cart using :ref:`clearEcommerceCart<jtc-api-clearEcommerceCart>` and register all items from the cart before tracking cart update, e.g.
-
-.. code-block:: javascript
+Tracking a cart update has two steps:
```

Comment:
Are those really `steps`? Wouldn't `modes` be a better word to describe this situation?

Reply:
I think it's steps because ideally you want to implement both. Modes for me implies that you can use one or the other but not both (usually).